nemo_skills/evaluation/top_answer_processor.py

```diff
@@ -0,0 +1,314 @@
+    # 2. extract: identify the best answer from input_files
+    mode: str = MISSING
+
+    # where to put the majority answer. By default replacing the expected_answer (assuming it's unknown)
+    # but change to predicted_answer, to follow up with a judge evaluation
+    fill_key: str = "expected_answer"
+
+    # if True, will not change the fill_key if it's already filled with not None
+    ignore_if_not_none: bool = False
+
+    # if True, will use string match to fill is_correct key
+    fill_is_correct: bool = True
+
+    # if True, use the highest scoring answer from the RM as the majority answer
+    use_highest_rm_score: bool = False
+
+    # if True, will use the RM score for weighted majority voting
+    use_majority_rm_score: bool = False
+
+    # if provided, will fail if can't find this many files. Useful in scheduled
+    # pipelines to ensure this step doesn't run if some of the expected files are missing
+    require_num_files: int | None = None
+
+    def __post_init__(self):
+        """Building data_file from dataset/split if not provided directly."""
+        if isinstance(self.input_files, str):
+            self.input_files = self.input_files.split(" ")
+
+
+cs = hydra.core.config_store.ConfigStore.instance()
+cs.store(name="base_process_top_answer_config", node=ProcessTopAnswerConfig)
+
+
+def map_to_output_path(file_path, input_dir, output_dir):
+    """Map the input file path to the output file path"""
+    # Convert all to Path objects
+    file_path, input_dir, output_dir = Path(file_path), Path(input_dir), Path(output_dir)
+
+    # Get the relative path from input_dir to the file
+    relative_path = file_path.relative_to(input_dir)
+
+    # Combine output_dir with the relative path
+    output_path = output_dir / relative_path
+
+    # Create parent directories if they don't exist
+    output_path.parent.mkdir(parents=True, exist_ok=True)
+
+    return output_path
+
+
+class ProcessMode(Enum):
+    FILL = "fill"
+    EXTRACT = "extract"
+
+
+class TopAnswerProcessor:
+    def __init__(self, cfg: ProcessTopAnswerConfig):
+        self.cfg = cfg
+        self._validate_cfg()
+
+    def _validate_cfg(self):
+        """Validate the config"""
+        cfg = self.cfg
+        # Check if the mode is valid
+        try:
+            self.process_mode = ProcessMode(cfg.mode)
+        except ValueError:
+            raise ValueError(f"Invalid mode: {cfg.mode}")
+        
+        # For fill mode, output_dir is required
+        if self.process_mode == ProcessMode.FILL:
+            if cfg.output_dir is None:
+                raise ValueError("output_dir is required when mode is fill")
+        
+        # Check whether the input_dir and output_dir are valid
+        if not Path(cfg.input_dir).exists():
+            raise ValueError(f"Input directory does not exist: {cfg.input_dir}")
+        if cfg.output_dir is not None and not Path(cfg.output_dir).exists():
```

Comment:
I think we should create output_dir in this case. All of our other scripts work like that

Reply:
Let me make that change